R/opts.R

```diff
@@ -0,0 +1,690 @@
+
+  # Perform input object validation
+  stop_if_not_gt(data = data)
+
+  # Ensure that all named locations are valid
+  if (!all(locations %in% c("column_labels", "stub", "row_group"))) {
+    stop("The available locations are `column_labels`, `stub`, and `row_group`.",
+         call. = FALSE)
+  }
+
+  if ("column_labels" %in% locations) {
+
+    data <-
+      tab_options(
+        data,
+        column_labels.text_transform = "uppercase",
+        column_labels.font.size = pct(80),
+        column_labels.font.weight = "bolder"
+      )
+  }
+
+  if ("stub" %in% locations) {
+
+    data <-
+      tab_options(
+        data,
+        stub.text_transform = "uppercase",
+        stub.font.size = pct(80),
+        stub.font.weight = "bolder"
+      )
+  }
+
+  if ("row_group" %in% locations) {
+
+    data <-
+      tab_options(
+        data,
+        row_group.text_transform = "uppercase",
+        row_group.font.size = pct(80),
+        row_group.font.weight = "bolder"
+      )
+  }
+
+  data
+}
+
+#' Option to remove the top and bottom borders
+#'
+#' The top and bottom borders of a **gt** table can be removed entirely by
+#' using the `opt_remove_top_bottom_borders()` function. The definition of what
+#' constitutes a top and bottom border depends on which table parts are present
+#' at the top and bottom (e.g., header, table columns, a footnote block, etc.).
+#' Regardless of the available parts, there won't be any top of bottom
+#' horizontal lines after using this function. This function serves as a
+#' convenient shortcut for `<gt_tbl> %>%
+#' tab_options(table.border.top.style = "hidden") %>%
+#' tab_options(table.border.bottom.style = "hidden")`.
+#'
+#' @inheritParams fmt_number
+#'
+#' @return An object of class `gt_tbl`.
+#'
+#' @examples
+#' # Use `exibble` to create a gt table with
+#' # a number of table parts added; next, we
+#' # remove the top and bottom borders with
+#' # `opt_remove_top_bottom_borders()`
+#' tab_1 <-
+#'   exibble %>%
+#'   gt(rowname_col = "row", groupname_col = "group") %>%
+#'   summary_rows(
+#'     groups = "grp_a",
+#'     columns = vars(num, currency),
+#'     fns = list(
+#'       min = ~min(., na.rm = TRUE),
+#'       max = ~max(., na.rm = TRUE)
+#'     )) %>%
+#'   grand_summary_rows(
+#'     columns = vars(currency),
+#'     fns = list(
+#'       total = ~sum(., na.rm = TRUE)
+#'     )) %>%
+#'   tab_source_note(source_note = "This is a source note.") %>%
+#'   tab_footnote(
+#'     footnote = "This is a footnote.",
+#'     locations = cells_body(columns = 1, rows = 1)
+#'   ) %>%
+#'   tab_header(
+#'     title = "The title of the table",
+#'     subtitle = "The table's subtitle"
+#'   ) %>%
+#'   opt_remove_top_bottom_borders()
+#'
+#' @section Figures:
+#' \if{html}{\figure{man_opt_remove_top_bottom_borders_1.svg}{options: width=100\%}}
+#'
+#' @family Table Option Functions
+#' @section Function ID:
+#' 9-5
+#'
+#' @export
+opt_remove_top_bottom_borders <- function(data) {
```

Comment:
Remove this function--currently it doesn't do anything important that `opt_remove_table_outline` doesn't do

Reply:
Now removed.